### DISCUSSION TITLE
Shelly HomeKit 2.12.1

### DISCUSSION THREAD

**Discussion by markirb** [mongoose-os-apps/shelly-homekit#1413] [Announcements]
## What's Changed

First release in a long time :)

Thanks to all contributors, testers and supporters.

If you use and enjoy the firmware please consider a [donation](https://www.paypal.com/donate/?hosted_button_id=RVFA9G5VMXRX8) to cover for hardware and hours lost along the way.

## Short Version

this is the condensed version, for contributors and details please see long version below

### New Model Support
* Plus 2PM
* Plus Plug S (without LED support)
* Plus Addon support for all supported Plus Models

### New Features
* Web UI Dark Mode
* HomeKit Adaptive Lighting for CCT Lights
* HomeKit Carbon dioxide and Monoxide sensors
* DHT Sensor Support for all Addons

### Fixes
* Upgradability from newer stock firmware versions
* PWM behavior at 100% for lights
* Uni: use Noisy Inputpin for debouncing Input
* Various small fixes see below

## Long
* Align top logo header by @web-premiere in https://github.com/mongoose-os-apps/shelly-homekit/pull/1060
* Introduce CSS variables by @ladvoc in https://github.com/mongoose-os-apps/shelly-homekit/pull/1083
* Support Dark Mode & UI issues fix by @Idomo in https://github.com/mongoose-os-apps/shelly-homekit/pull/1145
* require confirmation on password change by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1154
* ShellyUni: use noisyinputpin for debouncing by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1160
* code format fix for github actions clang-format by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1176
* use custom build of mgos/esp32-build docker image by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1192
* improve rgbw and cct labeling by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1195
* improve power meter by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1196
* [Script] fix wifi info for stock and pre 2.11.0 fw by @andyblac in https://github.com/mongoose-os-apps/shelly-homekit/pull/1003
* adaptive lighting support for CCT lights by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1028
* implement carbon dioxide and monoxide sensors by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1210
* option to invert hap state by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1224
* make nameserver configurable by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1206
* gdo option to use second output by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1205
* fix pwm behavior at 100% by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1227
* fix code gen with non alphanum chars ssid by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1226
* implement api toggle for switch and lightbulb by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1235
* Fix complaint when zeroconf returns props with a value of None by @pmkane in https://github.com/mongoose-os-apps/shelly-homekit/pull/1358
* Fix for stock upgradability on Plus Models; Github Workflow by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1387
* Addon unification / DHT support / addon as switch / plus addon support by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1340
* refactor supported device list by @timoschilling in https://github.com/mongoose-os-apps/shelly-homekit/pull/1190
* Plus2PM Support by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1388
* Add basic support for Shelly Plus Plug S by @trucidare in https://github.com/mongoose-os-apps/shelly-homekit/pull/1305
* Check DHT Sensor values for errors by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1392
* Plusaddon/digital in by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1396

## New Contributors
* @web-premiere made their first contribution in https://github.com/mongoose-os-apps/shelly-homekit/pull/1060
* @ladvoc made their first contribution in https://github.com/mongoose-os-apps/shelly-homekit/pull/1083
* @Idomo made their first contribution in https://github.com/mongoose-os-apps/shelly-homekit/pull/1145
* @pmkane made their first contribution in https://github.com/mongoose-os-apps/shelly-homekit/pull/1358
* @trucidare made their first contribution in https://github.com/mongoose-os-apps/shelly-homekit/pull/1305

**Full Changelog**: https://github.com/mongoose-os-apps/shelly-homekit/compare/2.11.2...2.12.1

<hr /><em>This discussion was created from the release <a href='https://github.com/mongoose-os-apps/shelly-homekit/releases/tag/2.12.1'>Shelly HomeKit 2.12.1</a>.</em>

**Comment by lieberjott** [comment]
A huge thanks to all contributors of this release, as this is a major step towards.

Many of us (especially those using Shelly 2.5 PMs for roller shutters and similar things) experience that their old Shellys are dying one after another and need be replaced with newer models (like 2 Plus PM) which had no Shelly Homekit support so far.
